Chessington.GameEngine/Pieces/Queen.cs

```diff
@@ -9,8 +9,145 @@ public Queen(Player player)
             : base(player) { }
 
         public override IEnumerable<Square> GetAvailableMoves(Board board)
-        {
-            return Enumerable.Empty<Square>();
+        { 
+            List<Square> availableMoves = new List<Square>();
+            
+            for (int i = board.FindPiece(this).Row + 1, j = board.FindPiece(this).Col + 1; i < 8 && j < 8; i++, j++)
+            {
+                if (board.GetPiece(new Square(i, j)) == null)
```

Comment:
You can extract the code inside the for loops into a function, because it looks the same, right?

Reply:
You could even extract it into a Piece class, since it's used by multiple pieces